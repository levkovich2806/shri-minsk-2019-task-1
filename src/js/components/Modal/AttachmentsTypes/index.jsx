import React from "react";
import styles from "./index.module.scss";

function AttachmentsTypes(props) {
  return (
    <div className={styles.types}>
      <div className={styles.types__title}>Приложения:</div>
      <div className={styles.types__item}>
        <input
          type="radio"
          value="link"
          id="att_type_link"
          name="attachmentType"
          onChange={props.onChange}
          defaultChecked
        />
        <label htmlFor="att_type_link">Ссылки</label>
      </div>
      <div className={styles.types__item}>
        <input
          type="radio"
          value="image"
          id="att_type_image"
          name="attachmentType"
          className={styles.types__item}
          onChange={props.onChange}
        />
        <label htmlFor="att_type_image">Изображения</label>
      </div>
    </div>
  );
}

export default AttachmentsTypes;
