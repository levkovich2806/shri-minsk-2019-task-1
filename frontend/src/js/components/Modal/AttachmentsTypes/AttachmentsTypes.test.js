import React from "react";
import { mount } from "enzyme";
import AttachmentsTypes from "./index.jsx";
import renderer from "react-test-renderer";

describe("AttachmentsTypes", function () {
  test("AttachmentsTypes rendering with type Link", () => {

    const props = {
      attachmentType: "link",
      onChange: () => { }
    }

    const component = renderer.create(<AttachmentsTypes {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("AttachmentsTypes rendering with type Image", () => {

    const props = {
      attachmentType: "image",
      onChange: () => { }
    }

    const component = renderer.create(<AttachmentsTypes {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
