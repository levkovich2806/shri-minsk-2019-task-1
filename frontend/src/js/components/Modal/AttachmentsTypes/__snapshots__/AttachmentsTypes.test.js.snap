// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentsTypes AttachmentsTypes rendering with type Image 1`] = `
<div
  className="types"
>
  <div
    className="types__title"
  >
    Приложения:
  </div>
  <div
    className="types__item"
  >
    <input
      checked={false}
      id="att_type_link"
      name="attachmentType"
      onChange={[Function]}
      type="radio"
      value="link"
    />
    <label
      htmlFor="att_type_link"
    >
      Ссылки
    </label>
  </div>
  <div
    className="types__item"
  >
    <input
      checked={true}
      className="types__item"
      id="att_type_image"
      name="attachmentType"
      onChange={[Function]}
      type="radio"
      value="image"
    />
    <label
      htmlFor="att_type_image"
    >
      Изображения
    </label>
  </div>
</div>
`;

exports[`AttachmentsTypes AttachmentsTypes rendering with type Link 1`] = `
<div
  className="types"
>
  <div
    className="types__title"
  >
    Приложения:
  </div>
  <div
    className="types__item"
  >
    <input
      checked={true}
      id="att_type_link"
      name="attachmentType"
      onChange={[Function]}
      type="radio"
      value="link"
    />
    <label
      htmlFor="att_type_link"
    >
      Ссылки
    </label>
  </div>
  <div
    className="types__item"
  >
    <input
      checked={false}
      className="types__item"
      id="att_type_image"
      name="attachmentType"
      onChange={[Function]}
      type="radio"
      value="image"
    />
    <label
      htmlFor="att_type_image"
    >
      Изображения
    </label>
  </div>
</div>
`;
